UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.SubscriptionService/Services/CallbackService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.Extensions.Logging;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using UKHO.ExternalNotificationService.Common.Logging;
+using UKHO.ExternalNotificationService.SubscriptionService.Helpers;
+
+namespace UKHO.ExternalNotificationService.SubscriptionService.Services
+{
+    public class CallbackService : ICallbackService
+    {
+        private readonly IAuthTokenProvider _authTokenProvider;
+        private readonly ILogger<CallbackService> _logger;
+        private readonly ICallbackClient _callbackClient;
+
+        public CallbackService(IAuthTokenProvider authTokenProvider, ILogger<CallbackService> logger, ICallbackClient callbackClient)
+        {
+            _authTokenProvider = authTokenProvider;
+            _logger = logger;
+            _callbackClient = callbackClient;
+        }
+
+        public async Task<HttpResponseMessage> CallbackToD365UsingDataverse(string externalEntityPath, object externalNotificationEntity, string d365CorrelationId, string correlationId)
+        {
+            string[] subscriptionId = externalEntityPath.Split("s(");
+            string subId = subscriptionId[1].Remove(subscriptionId[1].Length - 1);
```

Comment:
Can we pass subscriptionMessage object as parameter and then use properties directly instead of spilting subscriptionId from externalEntityPath?

Reply:
Done